File: code/cpp/ch15/practice/practice_05_dutch_national_flag.cpp
/*
 * Practice 5: Dutch National Flag
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 */
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

vector<int> solve(vector<int> arr) {
    // TODO: Replace this with your solution
    return arr;
}

int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    vector<int> result = solve(arr);
    for (int i = 0; i < (int)result.size(); i++) {
        if (i > 0) cout << " ";
        cout << result[i];
    }
    cout << endl;
    return 0;
}

File: code/java/ch15/practice/Practice05DutchNationalFlag.java
package ch15.practice;

import java.util.*;

/**
 * Practice 5: Dutch National Flag
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 *
 * PROBLEM: Sort an array of 0s, 1s, and 2s in one pass.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Practice05DutchNationalFlag {
    public static int[] solve(int[] arr) {
        // TODO: Replace this with your solution
        return arr;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine().trim();
        int[] arr = line.isEmpty() ? new int[]{} :
            Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
        System.out.println(Arrays.toString(solve(arr)));
        sc.close();
    }
}
